### DISCUSSION TITLE
Support opened local files

### DISCUSSION THREAD

**Discussion by gildas-lormeau** [gildas-lormeau/SingleFile#627] [Q&A]
link to original post: https://github.com/gildas-lormeau/SingleFile/issues/625

I'd like to see support for opened files.

My use case is thus: I use Calibre to convert my eBook to HTMLZ and then extract that as a zip file. I then add some CSS of my own and then use this extension as a simple way to get the external CSS stylesheet and images embedded into a single HTML file.

**Comment by gildas-lormeau** [comment]
Did you enable the option "Allow access to file URLs" in the extension page? e.g. chrome://extensions/?id=mpiodijhokgodhhofbcjdecpffjipkle in Chrome.

**Comment by JustinBiesboer** [comment]
I have local files to convert with the CLI that fail (local images).
Prior to setting the "allow access to file URLs" setting, the Chrome plugin did not work on these files.  Afterward it did.  CLI succeeds with chrome but results in broken images.  Haven't tried FF.
`<img class="default" src="data:null;base64,">`
Thanks!

**Reply by JustinBiesboer** [reply]
I've subsequently tried a few things.
1.  Many variations on the local file URLs.  All view correctly in chrome, but none work with the CLI.
2. jsdom and FF via selenium.  I get errors from these - not sure it's worth pursuing this path unless there's something fundamental about chrome that won't let the local files work.
3. CLI works fine with actual external/http image links.

**Reply by JustinBiesboer** [reply]
I got FF to work and jsdom to mostly work (jsdom does not convert an external png image, instead giving it `<img src="data:application/octet-stream;base64, ... >`).
For Chrome, I've tried passing in various browser args `--browser-args='[\"--allow-file-access-from-files\"]'`, but I'm not sure if I'm doing it right or those will give me the correct result anyway.

**Reply by JustinBiesboer** [reply]
Summary:

1. FF works with file:/// formatted links.  Cannot load images from insecure https.  Human-started browser works.  Selenium-started browser doesn't work (it seems to need the setting security.enterprise_roots.enabled = true in about:config), and I'm not sure how to get that enabled with SingleFile.
2. Same issue as above with jsdom.  Won't load image from insecure https.  Haven't dug into it very far.
2a. So far jsdom backend only works with a web server hosting the local files, rather than using local filesystem paths.  This is all totally new to me, so I'm fumbling through it.
4. Chrome also only works (properly) as long as there's a web server running so that links are not in local filesystem format.  Can't get chrome to succeed with filesystem URLs for the images (even though it all loads perfectly with human-started browser).
